src/socket-message-handling.ts

```diff
@@ -76,52 +159,40 @@ const setupWSConnection = (conn) => {
   conn.on(
     "message",
     /** @param {object} message */ (message) => {
-      if (typeof message === "string") {
-        message = JSON.parse(message);
-      }
-      if (message && message.type && !closed) {
-        switch (message.type) {
-          case "subscribe":
-            /** @type {Array<string>} */ (message.topics || []).forEach(
-              (topicName) => {
-                if (typeof topicName === "string") {
-                  // add conn to topic
-                  const topic = map.setIfUndefined(
-                    topics,
-                    topicName,
-                    () => new Set()
-                  );
-                  topic.add(conn);
-                  // add topic to conn
-                  subscribedTopics.add(topicName);
-                }
-              }
-            );
-            break;
-          case "unsubscribe":
-            /** @type {Array<string>} */ (message.topics || []).forEach(
-              (topicName) => {
-                const subs = topics.get(topicName);
-                if (subs) {
-                  subs.delete(conn);
-                }
-              }
-            );
-            break;
-          case "publish":
-            if (message.topic) {
-              const receivers = topics.get(message.topic);
-              if (receivers) {
-                receivers.forEach((receiver) => send(receiver, message));
-              }
-            }
-            break;
-          case "ping":
-            send(conn, { type: "pong" });
+      if (message && !closed) {
+        if (Buffer.isBuffer(message)) {
+          providerMessageResolver(conn, new Uint8Array(message), doc);
```

Comment:
the more I look at this the more I think we should split this higher up, at the `websocket-service` level. Since the client ends up creating two websocket connections anyway, split it by URL path might be the cleanest thing to do. Maybe something like this: 

```js

wss.on("connection-signaling", setupSignalingConnection);
wss.on("connection-provider", setupProviderConnection);

server.on("upgrade", (request, socket, head) => {
  if(request.url === "signal"){
    wss.handleUpgrade(request, socket, head, ()=>
      wss.emit("connection-signaling", ws, request)
    );
  } else {
    wss.handleUpgrade(request, socket, head, ()=>
      wss.emit("connection-provider", ws, request)
    );
  }
});
```

Reply:
And then have two different files, one for `SignalingConnection` things and the other for `ProviderConnection` stuff

Reply:
I like that way a lot, I'll work on that now